src/api/hooks/useGetRegistrationsWithDoi.ts

```diff
@@ -0,0 +1,10 @@
+import { useQuery } from '@tanstack/react-query';
+import { fetchResults } from '../searchApi';
+
+export const useGetRegistrationsWithDoi = (doiQuery: string) => {
+  return useQuery({
+    enabled: !!doiQuery,
+    queryKey: ['doi-results', doiQuery],
+    queryFn: () => fetchResults({ doi: doiQuery }),
+  });
+};
```

Comment:
Vi ønsker sikkert heller at denne hooken skal kunne brukes for alle resultatsøk-heller enn å lage en ny hook for hver gang?

```suggestion
export const useGetRegistrationsWithDoi = (params: FetchResultsParams) => {
  return useQuery({
    enabled: !!params,
    queryKey: ['registrations', params],
    queryFn: () => fetchResults(params),
  });
};
```

Reply:
Det blir laget en ny hook for hver komponent som bruker den uansett, men vi sparer selvsagt litt kode om vi gjør den mer generisk. Men ender vi ikke opp med å lagre et annet sted i cachen med koden din over?

Reply:
bare om de har ulike params, og det ønsker vi jo, siden det er ulike oppslag